sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationkeyvaultoptions.py

```diff
@@ -3,27 +3,53 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # -------------------------------------------------------------------------
-
+from typing import overload, List, Optional, Callable
+from azure.keyvault.secrets import SecretClient
 
 class AzureAppConfigurationKeyVaultOptions:
-    """
-    Options for connecting to Key Vault.
 
-    :param credential: A credential for authenticating with the key vault. This is optional if secret_clients is
-     provided.
-    :type credential: ~azure.core.credentials.TokenCredential
-    :param secret_clients: A list of SecretClient from azure-keyvault-secrets. This is optional if credential is
-     provided.
-    :type secret_clients: list[~azure.keyvault.secrets.SecretClient]
-    :param secret_resolver: A function that takes a URI and returns a value.
-    :type secret_resolver: callable
-    """
+    @overload
+    def __init__(self,  credential, *, secret_clients: Optional[List[SecretClient]] = None):
```

Comment:
So, the logic is this, if a secret client is provided for a store, we check it's vault url compared to the one in app config, than the value provided is used, otherwise it fallback on the other method, either a general credential or a `secret_resolver` as they are both default options you can't use both.